ngraph_bridge/ngraph_builder.cc

```diff
@@ -3378,46 +3378,48 @@ static Status TranslateTileOp(
   return Status::OK();
 }
 
-// // Translate TopKV2 Op using ngraph core op TopK
-// static Status TranslateTopKV2Op(
-//     const Node* op, const std::vector<const Tensor*>& static_input_map,
-//     Builder::OpMap& ng_op_map) {
-//   ng::Output<ngraph::Node> ng_input;
-
-//   TF_RETURN_IF_ERROR(ValidateInputCount(op, 2));
-//   TF_RETURN_IF_ERROR(GetInputNode(ng_op_map, op, 0, ng_input));
-
-//   size_t k_axis = ng_input.get_shape().size() - 1;
-//   std::vector<int32> ng_k;
-//   size_t k;
-//   bool sorted = true;
-//   TF_RETURN_IF_ERROR(GetStaticInputVector(op, 1, static_input_map, &ng_k));
-
-//   k = ng_k[0];
-//   // sorted = false is not supported right now, it falls back to TF if set to
-//   // false.
-//   TF_RETURN_IF_ERROR(GetNodeAttr(op->attrs(), "sorted", &sorted));
-
-//   // index element type - currently only int32 or int64 are supported by
-//   // ngraph
-//   auto ng_result = std::make_shared<ngraph::op::TopK>(ng_input, k_axis,
-//   ng::element::i32, k, sorted);
-//   Builder::SetTracingInfo(op->name(), ng_result);
-
-//   // ng::Output<ng::Node> ng_values = ng_result->output(1);
-//   // ng::Output<ng::Node> ng_indices = ng_result->output(0);
-//  auto ng_values = std::make_shared<ngraph::op::GetOutputElement>(ng_result,
-//  1);
-//  Builder::SetTracingInfo(op->name(), ng_values);
-//  auto ng_indices = std::make_shared<ngraph::op::GetOutputElement>(ng_result,
-//  0);
-//  Builder::SetTracingInfo(op->name(), ng_indices);
-
-//   SaveNgOp(ng_op_map, op->name(), ng_values);
-//   SaveNgOp(ng_op_map, op->name(), ng_indices);
-
-//   return Status::OK();
-// }
+// Translate TopKV2 Op using ngraph core op TopK
+static Status TranslateTopKV2Op(
+    const Node* op, const std::vector<const Tensor*>& static_input_map,
+    Builder::OpMap& ng_op_map) {
+  ng::Output<ngraph::Node> ng_input;
+
+  TF_RETURN_IF_ERROR(ValidateInputCount(op, 2));
+  TF_RETURN_IF_ERROR(GetInputNode(ng_op_map, op, 0, ng_input));
+
+  // axis along which to compute top k indices
+  int64 k_axis = ng_input.get_shape().size() - 1;
+
+  // scalar input tensor specifying how many max/min elts should be computed
+  std::vector<int64> ng_k_vec;
+  TF_RETURN_IF_ERROR(GetStaticInputVector(op, 1, static_input_map, &ng_k_vec));
+  auto ng_k = ConstructNgNode<opset::Constant>(op->name(), ng::element::i64,
```

Comment:
```suggestion
  auto ng_k = ConstructNgNode<opset::Constant>(op->name(), ng::element::i32,
```

Reply:
https://github.com/NervanaSystems/ngraph/blob/master/src/ngraph/op/topk.cpp#L155
ngraph only supports i64 for k as element type

Reply:
That's for v0.